config/initializers/saml_idp.rb

```diff
@@ -0,0 +1,60 @@
+require 'service_provider'
+
+SamlIdp.configure do |config|
+  protocol = Rails.env.development? ? 'http://' : 'https://'
+  api_base = protocol + Figaro.env.domain_name + '/api'
+
+  config.x509_certificate = Rails.application.secrets.saml_cert
+  config.secret_key = OpenSSL::PKey::RSA.new(
+    File.read(Rails.root + 'config/saml.key.enc'),
+    Figaro.env.saml_passphrase).to_pem
+
+  # config.signature_alg = 'rsa-sha256'
+  # config.digest_alg = 'sha256'
+
+  # Disabled because authnrequest signing doesn't prevent any
+  # attack that can be thought of.
+  # config.verify_authnrequest_sig = true
```

Comment:
We at least have to verify the request at some point and I think the sig is the best way to go. I don't understand how we prevent the user from being redirect to a malicious URL (along with whatever attribs we include in the `SAMLResponse`) in a seemingly valid authn request?

I agree that this is ripe for phishing since most email scanners (ie. gmail or org/enterprise solutions like barracuda/ironport) will not flag a malicious link since it's no different than existing authn requests and it points to a trusted domain. Furthermore the user may gain false trust since they are first authenticating with a known site and then are probably less likely to pay attention to the URL after that point. That said, I'm aware of a few .gov sites that have been operating like this in production for quite some time so perhaps it's not as big of a problem as it seems.